src/ghci/mod.rs

```diff
@@ -313,7 +313,7 @@ impl Ghci {
             for path in add {
                 let add_result = this.lock().await.add_module(path).await?;
                 if let Some(CompilationResult::Err) = add_result {
-                    compilation_failed = false;
+                    compilation_failed = true;
```

Comment:
Oops

Reply:
oh yeah, i forgot about this, too.  in fact, this was the line of code that made me ask about the result type refactor because i wasn't sure if i understood the type correctly